src/controller_examples/rabbitmq_controller/exec/resource/default_user_secret.rs

```diff
@@ -0,0 +1,121 @@
+        let sts = Secret::unmarshal(obj);
+        if sts.is_ok() {
+            Ok(sts.unwrap())
+        } else {
+            Err(RabbitmqError::Error)
+        }
+    }
+
+    fn state_after_create_or_update(obj: DynamicObject, state: RabbitmqReconcileState) -> (res: Result<RabbitmqReconcileState, RabbitmqError>) {
+        let secret = Secret::unmarshal(obj);
+        if secret.is_ok() {
+            Ok(RabbitmqReconcileState {
+                reconcile_step: RabbitmqReconcileStep::AfterKRequestStep(ActionKind::Get, ResourceKind::PluginsConfigMap),
+                ..state
+            })
+        } else {
+            Err(RabbitmqError::Error)
+        }
+    }
+
+    fn next_resource_get_request(rabbitmq: &RabbitmqCluster) -> (res: Option<KubeGetRequest>) {
+        Some(PluginsConfigMapBuilder::get_request(rabbitmq))
+    }
+}
+
+pub fn update_default_user_secret(rabbitmq: &RabbitmqCluster, found_secret: Secret) -> (secret: Secret)
+    requires
+        rabbitmq@.metadata.name.is_Some(),
+        rabbitmq@.metadata.namespace.is_Some(),
+    ensures
+        secret@ == spec_resource::update_default_user_secret(rabbitmq@, found_secret@),
+{
+    let mut user_secret = found_secret.clone();
+    let made_user_secret = make_default_user_secret(rabbitmq);
+    // TODO: whether to update ports
+    user_secret.set_metadata({
+        let mut metadata = found_secret.metadata();
+        metadata.set_owner_references(make_owner_references(rabbitmq));
+        metadata.unset_finalizers();
+        metadata.set_labels(made_user_secret.metadata().labels().unwrap());
+        metadata.set_annotations(made_user_secret.metadata().annotations().unwrap());
+        metadata
+    });
+    user_secret
+}
+
+pub fn make_default_user_secret_name(rabbitmq: &RabbitmqCluster) -> (name: String)
+    requires
+        rabbitmq@.metadata.name.is_Some(),
+        rabbitmq@.metadata.namespace.is_Some(),
+    ensures
+        name@ == spec_resource::make_default_user_secret_name(rabbitmq@),
+{
+    rabbitmq.name().unwrap().concat(new_strlit("-default-user"))
+}
+
+pub fn make_default_user_secret(rabbitmq: &RabbitmqCluster) -> (secret: Secret)
+    requires
+        rabbitmq@.metadata.name.is_Some(),
+        rabbitmq@.metadata.namespace.is_Some(),
+    ensures
+        secret@ == spec_resource::make_default_user_secret(rabbitmq@)
+{
+    let mut data = StringMap::empty();
+    data.insert(new_strlit("username").to_string(), new_strlit("user").to_string());
+    data.insert(new_strlit("password").to_string(), new_strlit("changeme").to_string());
+    data.insert(new_strlit("type").to_string(), new_strlit("rabbitmq").to_string());
+    data.insert(new_strlit("host").to_string(),
+            rabbitmq.name().unwrap().concat(new_strlit(".")).concat(rabbitmq.namespace().unwrap().as_str()).concat(new_strlit(".svc"))
+    );
+    data.insert(new_strlit("provider").to_string(), new_strlit("rabbitmq").to_string());
+    // TODO: check \n
```

Comment:
What is this TODO?

Reply:
Because I am not sure if there is a line break in the end of `"default_user = user\ndefault_pass = changeme"`